### PR TITLE
feat: separate freq tols for truth and tracking oracles

### PR DESCRIPTION
**Pull request by chad-js**

### CODE REVIEW

File: src/ChainlinkTriggerFactory.sol [hunk 1]
```diff
@@ -102,7 +108,10 @@ contract ChainlinkTriggerFactory is IChainlinkTriggerFactoryEvents {
     uint256 _frequencyTolerance
   ) public returns (ChainlinkTrigger _trigger) {
     AggregatorV3Interface _truthOracle = deployFixedPriceAggregator(_price, _decimals);
-    return deployTrigger(_truthOracle, _trackingOracle, _priceTolerance, _frequencyTolerance);
+
+    // For the truth FixedPriceAggregator peg oracle, we use a frequency tolerance of 1 since it should always return
+    // block.timestamp as the updatedAt timestamp.
+    return deployTrigger(_truthOracle, _trackingOracle, _priceTolerance, 1, _frequencyTolerance);
```
Comment (davidlaprade) [inline src/ChainlinkTriggerFactory.sol]: 💯 This is exactly what I was looking for, nice

File: test/ChainlinkTriggerFactory.t.sol [hunk 2]
```diff
@@ -124,22 +126,27 @@ contract DeployTriggerForkTest is ChainlinkTriggerFactoryTestBaseSetup {
...
-        _frequencyTolerance
+        _trackingFrequencyTolerance
       );
       AggregatorV3Interface _pegOracle = _trigger.truthOracle();
       (,int256 _priceInt,,,) = _pegOracle.latestRoundData();
       assertEq(_priceInt, _pegPrice);
       assertEq(_pegOracle.decimals(), _pegDecimals);
       _truthOracle = address(_pegOracle);
+
+      // For peg triggers, we set the frequency tolerance to 0 for the truth FixedPriceAggregator peg oracle.
```
Comment (davidlaprade) [inline test/ChainlinkTriggerFactory.t.sol:148]:
```suggestion
      // For peg triggers, we set the frequency tolerance to 1 for the truth FixedPriceAggregator peg oracle.
```

File: test/ChainlinkTrigger.t.sol [hunk 3]
```diff
@@ -202,11 +206,12 @@ contract ProgrammaticCheckTest is ChainlinkTriggerUnitTest {
 
     vm.warp(_currentTimestamp);
     if (
-      _truthOracleUpdatedAt + frequencyTolerance < block.timestamp ||
-        _targetOracleUpdatedAt + frequencyTolerance < block.timestamp
+      _truthOracleUpdatedAt + truthFrequencyTolerance < block.timestamp ||
+        _targetOracleUpdatedAt + trackingFrequencyTolerance < block.timestamp
```
Comment (davidlaprade) [inline test/ChainlinkTrigger.t.sol:210]: Nit: it'd be great if we could fuzz over the frequency tolerances for these oracles too. Not blocking though

**Review by davidlaprade — APPROVED** [review]
Looks great, thanks for doing this!

File: src/ChainlinkTriggerFactory.sol (hunk 1 above)
Comment (mds1) [inline src/ChainlinkTriggerFactory.sol]: This should be zero right? Since it's the same block they'll always match

Comment (chad-js) [reply src/ChainlinkTriggerFactory.sol]: oops, definitely should be. I had it as that before but tests were still failing, so I figured some weird test env issue but now it works. I must've missed something earlier

Comment (chad-js) [reply src/ChainlinkTriggerFactory.sol]: [afc57b3](https://github.com/Cozy-Finance/cozy-v2-triggers/pull/4/commits/afc57b3df4e5a6f10657e5623fa5f4f7d69bf635)

File: test/ChainlinkTriggerFactory.t.sol (hunk 2 above)
Comment (chad-js) [reply test/ChainlinkTriggerFactory.t.sol:148]: will leave as-is per https://github.com/Cozy-Finance/cozy-v2-triggers/pull/4#discussion_r944602469

File: test/ChainlinkTrigger.t.sol (hunk 3 above)
Comment (chad-js) [reply test/ChainlinkTrigger.t.sol:210]: this test fuzzes over `_truthOracleUpdatedAt` and `_targetOracleUpdatedAt`, so I think this test is sufficient, gonna leave this as-is

**Review by mds1 — APPROVED** [review]

**Pull request merged**